cognite/neat/graph/extractors/_classic_cdf/_assets.py

```diff
@@ -21,17 +22,28 @@ class AssetsExtractor(BaseExtractor):
         namespace (Namespace, optional): The namespace to use. Defaults to DEFAULT_NAMESPACE.
         to_type (Callable[[Asset], str | None], optional): A function to convert an asset to a type. Defaults to None.
             If None or if the function returns None, the asset will be set to the default type "Asset".
+        total (int, optional): The total number of assets to load. If passed, you will get a progress bar if rich
+            is installed. Defaults to None.
+        limit (int, optional): The maximal number of assets to load. Defaults to None. This is typically used for
+            testing setup of the extractor. For example, if you are extracting 100 000 assets, you might want to
+            limit the extraction to 1000 assets to test the setup.
     """
 
+    _SPACE_PATTERN = re.compile(r"\s+")
```

Comment:
As values might contain non-alpha num characters that can ill form URI, suggestion to remove any character that will not fit end View/Container id:

r"(^[a-zA-Z][a-zA-Z0-9_]{0,253}[a-zA-Z0-9]?$)"

We have in :
https://github.com/cognitedata/neat/blob/main/cognite/neat/rules/models/_types/_base.py#L4C1-L4C25

Various compliance regexes (which I guess should be move to utils)